packages/core/EventManager.ts

```
@@ -400,20 +404,32 @@ export default class EventManager {
             (c) => c.type === destination.integration
           );
           // It might not be the first connected calendar as it seems that the order is not guaranteed to be ascending of credentialId.
-          const firstCalendarCredential = destinationCalendarCredentials[0];
-          log.warn(
-            "No credentialId found for destination calendar, falling back to first found calendar",
-            safeStringify({
-              destination: getPiiFreeDestinationCalendar(destination),
-              firstConnectedCalendar: getPiiFreeCredential(firstCalendarCredential),
-            })
-          );
-
-          createdEvents.push(await createEvent(firstCalendarCredential, event));
+          const firstCalendarCredential: (typeof destinationCalendarCredentials)[number] | undefined =
+            destinationCalendarCredentials[0];
+          // It is seen in the production that there are `destinationCalendars` with no calendar credential at all.
+          if (firstCalendarCredential) {
+            loggerWithEventDetails.warn(
+              `No credentialId found for destination calendar, falling back to first found credential matching  destination.integration=${destination.integration}`,
+              safeStringify({
+                destination: getPiiFreeDestinationCalendar(destination),
+                firstConnectedCalendar: getPiiFreeCredential(firstCalendarCredential),
+              })
+            );
+
+            createdEvents.push(await createEvent(firstCalendarCredential, event));
+          } else {
+            loggerWithEventDetails.error(
```

Log the erroneous situation which was earlier silently ignored.